js/build.js

```diff
@@ -62,11 +72,95 @@ function init(){
 
   });
 }
+
+// Cheking if user changed array order in interface
+function hasArrayChangedOrder (domArray, newArray) {
+  var changed = false;
+
+  domArray.each(function(index, element) {
+    if ( element.dataset.sliderId !== newArray[index].id ) {
+      changed = true;
+      return false;
+    }
+  });
+
+  return changed;
+}
+
+// Drawing slides according to new array order
+function reDrawAllSlides(data, widgetId) {
+  
+  globalSwiper[widgetId].removeAllSlides();
+
+  _.forEach(data, function (item) {
+    globalSwiper[widgetId].appendSlide('<div class="swiper-slide" data-slider-id="'+item.id+'"></div>');
+  });
+}
+
+// Main fanction to update and show slides
+function updateSlide (data, widgetId, activeSlide) {
+
+  var $slidesInDom = $('[data-onboarding-id='+widgetId+'] .swiper-container [data-slider-id]');
+
+  // Reload widget build only if we deleted all slides or we init new slider on the same screen or after we deleted all slides and start to add the again
+  if (!globalSwiper[widgetId] || !data.length || deletedAllSlides) {
+    Fliplet.Studio.emit('reload-widget-instance', widgetId);
+    deletedAllSlides = data.length === 0;
+    return;
+  }
+
+  var currentSlide = activeSlide !== undefined ? activeSlide : globalSwiper[widgetId].activeIndex;
+  
+  if ($slidesInDom.length !== data.length) {
+    if ($slidesInDom.length > data.length) {
+      var deletedPosition;
+      $slidesInDom.each(function(index, element) {
+        if (!data[index] || element.dataset.sliderId !== data[index].id) {
+          globalSwiper[widgetId].removeSlide(index);
+          deletedPosition = index;
+          return false;
+        }
+      });
+      if (deletedPosition === ($slidesInDom.length - 1)) {
+        currentSlide = deletedPosition - 1;
+      } else {
+        currentSlide = deletedPosition;
+      }
+    } else {
+      globalSwiper[widgetId].appendSlide('<div class="swiper-slide" data-slider-id="'+data[data.length-1].id+'"></div>');
+      currentSlide = data.length - 1;
+    }
+  } else { 
+    var arrayChanged = hasArrayChangedOrder($slidesInDom, data);
+    if (arrayChanged) {
```

Comment:
No need for an additional variable

```suggestion
    if (hasArrayChangedOrder($slidesInDom, data)) {
```